bigtable/client/row_range.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/row_range.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace btproto = ::google::bigtable::v2;
+
+bool RowRange::IsEmpty() const {
+  std::string unused;
+  // We do not want to copy the strings unnecessarily, so initialize a reference
+  // pointing to *_key_closed() or *_key_open(), as needed.
+  std::reference_wrapper<const std::string> start(unused);
```

Comment:
Is this a fancy way of avoiding to use pointers to strings (and using references instead)? :-)

Reply:
Do not look under the hood, those might be implemented as pointers too :-)  Happy to change if you think it would be more readable.

Reply:
I _think_ using pointers here would be simpler and easier to understand; I think the value of `reference_wrapper` and `cref` is good for putting references in containers, but here I think it overcomplicates what is really just a pointer to one or another string; I had to look this up to make sure I was understanding what it was doing, and it seems too complex for the task at hand.

Reply:
Done.